### PR TITLE
refactoring and improvements

### PR DESCRIPTION
**Pull request by squirrelsc**
A couple of refactoring, see details for each commits. Some refactoring is for next in parallel task.

### CODE REVIEW

File: examples/testsuites/helloworld.py [hunk 1]
```diff
@@ -17,6 +18,7 @@
     this is an example test suite.
     it helps to understand how to write a test case.
     """,
+    requirement=simple_requirement(unsupported_os=[]),
```
Comment (sharsonia) [inline examples/testsuites/helloworld.py:21]: Do we want some OS not to have support to even run basic LISA test?

File: lisa/sut_orchestrator/azure/features.py [hunk 2]
```diff
@@ -35,7 +38,8 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
         self._initialize_information(self._node)
 
     def _execute(self, wait: bool, operator: str) -> Any:
```
Comment (sharsonia) [inline lisa/sut_orchestrator/azure/features.py:40]: Why we returning Any here? we can just have None?

File: examples/testsuites/helloworld.py (hunk 1 above)
Comment (squirrelsc) [reply examples/testsuites/helloworld.py:21]: So far, there is no such OS discovered. The hello world examples are very simple, it should be able to run in any situation. If they cannot be run, even the OS detection doesn't work.

Comment (sharsonia) [reply examples/testsuites/helloworld.py:21]: Then what's the point of having this list if we don't have any such OS yet.

Comment (squirrelsc) [reply examples/testsuites/helloworld.py:21]: Because the default value is [Windows] now, if we don't set it's an empty list.

**Pull request merged**